	Patches for pdfTeX / luaTeX / XeTeX

	Declare GlobalParams constructor without config file argument
	and skip reading a config file.

	from:
		Han The Thanh <thanh@pdftex.org>
		Peter Breitenlohner <tex-live@tug.org>

diff -ur -N xpdf-3.02.orig/xpdf/GlobalParams.cc xpdf-3.02/xpdf/GlobalParams.cc
--- xpdf-3.02.orig/xpdf/GlobalParams.cc	2009-04-02 10:22:18.000000000 +0200
+++ xpdf-3.02/xpdf/GlobalParams.cc	2009-05-10 13:01:34.000000000 +0200
@@ -608,10 +608,15 @@
 // parsing
 //------------------------------------------------------------------------
 
+#ifdef PDF_PARSER_ONLY
+GlobalParams::GlobalParams() {
+  UnicodeMap *map;
+#else
 GlobalParams::GlobalParams(char *cfgFileName) {
   UnicodeMap *map;
   GString *fileName;
   FILE *f;
+#endif
   int i;
 
 #if MULTITHREADED
@@ -755,6 +760,7 @@
   map = new UnicodeMap("UCS-2", gTrue, &mapUCS2);
   residentUnicodeMaps->add(map->getEncodingName(), map);
 
+#ifndef PDF_PARSER_ONLY
   // look for a user config file, then a system-wide config file
   f = NULL;
   fileName = NULL;
@@ -792,6 +798,7 @@
     delete fileName;
     fclose(f);
   }
+#endif /* !PDF_PARSER_ONLY */
 }
 
 void GlobalParams::createDefaultKeyBindings() {
diff -ur -N xpdf-3.02.orig/xpdf/GlobalParams.h xpdf-3.02/xpdf/GlobalParams.h
--- xpdf-3.02.orig/xpdf/GlobalParams.h	2007-02-27 23:05:52.000000000 +0100
+++ xpdf-3.02/xpdf/GlobalParams.h	2009-05-10 13:01:46.000000000 +0200
@@ -5,6 +5,9 @@
 // Copyright 2001-2003 Glyph & Cog, LLC
 //
 //========================================================================
+//  Modified for TeX Live by Peter Breitenlohner <tex-live@tug.org>
+//  See top-level ChangeLog for a list of all modifications
+//========================================================================
 
 #ifndef GLOBALPARAMS_H
 #define GLOBALPARAMS_H
@@ -192,9 +195,14 @@
 class GlobalParams {
 public:
 
+#ifdef PDF_PARSER_ONLY
+  // Initialize the global parameters  without reading a config file.
+  GlobalParams();
+#else
   // Initialize the global parameters by attempting to read a config
   // file.
   GlobalParams(char *cfgFileName);
+#endif
 
   ~GlobalParams();
 
